pkg/cmd/init.go

```diff
@@ -398,3 +397,36 @@ func initKubeconfig(
 	}
 	return nil
 }
+
+// certsToRegenerate returns paths to certificates in the given certificate chains
+// bundle that need to be regenerated
+func certsToRegenerate(cs *certchains.CertificateChains) ([][]string, error) {
+	regenCerts := [][]string{}
+	err := cs.WalkChains(nil, func(certPath []string, c x509.Certificate) error {
+		if now := time.Now(); now.Before(c.NotBefore) || now.After(c.NotAfter) {
+			regenCerts = append(regenCerts, certPath)
+		}
+
+		timeLeft := time.Until(c.NotAfter)
+
+		const month = 30 * time.Hour * 24
+
+		if cryptomaterial.IsCertShortLived(&c) {
+			// the cert has less than 7 months to live, just rotate
+			until := 7 * month
+			if timeLeft < until {
+				regenCerts = append(regenCerts, certPath)
+			}
+			return nil
+		}
+
+		// long lived certs
+		if timeLeft < 18*month {
+			regenCerts = append(regenCerts, certPath)
+		}
+
+		return nil
+	})
+
+	return regenCerts, err
```

Comment:
I wonder whether we may want to just regenerate all the certs of a class (ShortLived/LongLived) if one of the certs in the class has reached the point of rotation :thinking: